source/server/http/admin.cc

```diff
@@ -275,6 +276,36 @@ Http::Code AdminImpl::handlerClusters(const std::string&, Http::HeaderMap&,
   return Http::Code::OK;
 }
 
+// TODO(jsedgwick) Use query params to list available dumps, selectively dump, etc
+Http::Code AdminImpl::handlerConfigDump(const std::string&, Http::HeaderMap&,
+                                        Buffer::Instance& response) const {
+  envoy::admin::ConfigDump dump;
+  auto& config_dump_map = *(dump.mutable_configs());
+  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+    ProtobufTypes::MessagePtr message;
+    try {
+      message = key_callback_pair.second();
+    } catch (const std::exception& e) {
+      ENVOY_LOG(warn, "Config dump callback with key \"{}\" threw an exception: {}",
+                key_callback_pair.first, e.what());
+      continue;
+    }
+
+    if (!message) {
```

Comment:
See above, except even softer case since they're just failing to follow an API, not letting an exception escape. I am just picturing everything humming along fine and then someone is firefighting and consuming admin and crashes all their instances because it turns out their logging function didn't return the right thing.

Reply:
Yup that's a bug and should be tested and fixed. I would fail hard here for now. RELEASE_ASSERT if you like over normal ASSERT.